table/block_based/filter_policy.cc

```diff
@@ -168,10 +199,24 @@ class XXPH3FilterBitsBuilder : public BuiltinFilterBitsBuilder {
       *aggregate_rounding_balance_ += diff;
     } else {
       buf->reset(new char[rv]());
+      // Reserve cache for buf
+      if (cache_res_mgr) {
+        Status s = cache_res_mgr->UpdateCacheReservation<
```

Comment:
You have this pattern repeating all over the place, making for
* A large mass of code for an auxiliary feature
* Difficulty connecting reservation increase point to corresponding reservation decrease point (when applicable?)
* Potential inaccuracy when ignoring status (decrease missing corresponding increase)
* The smell of a threading data race between GetTotalMemoryUsed and UpdateCacheReservation. (I don't think we're dealing with multiple threads here, but it looks suspicious.)

I think it would be easier to understand what was going on if each incremental reservation had its own RAII handle (possibly Cache::Handle) that releases the reservation when released. The FilterBitsBuilder API doesn't make that way easy but I think that's the approach we should be looking at.